### ISSUE TITLE
Catalog Rule & Selected Categories with level > 3

### ISSUE THREAD

**Issue by U12358** [magento/magento2#17493]
### Preconditions
1. M 2.1,4 

### Steps to reproduce
1. Open form for create new Catalog Rule 
2. In Condition Select^ Product Attribute -> Category
3. Open Categories Chooser
4. Select one Category with level = 1, 2 or 3
5. Select one Category with level > 3 (4, 5, ...)
6. Click ok & close Category Chooser
5. Open again Category Chooser and see
   that only categories with level <= 3 are selected
   and category with level > 3 without selection

### Expected result
1. All selected priviosly categories must be selected, with level > 3

### Actual result
1. Categories with level > 3 are not selected

### PS
I used M2.1.,4, but checked 2.3 version and this bug is still present.
Categories with level <= 3 are generated with page with level > 3 with ajax request
For myself I fixed in file Catalog/view/adminhtml/templates/catalog/category/checkboxes/tree.phtml
2. Added selected param to TreeLoader
3. Change Controller from CatalogRule/Controller/Adminhtml/Promo/Widget/CategoriesJson.php
to Catalog/Controller/Adminhtml/Category/Widget/CategoriesJson.php
which can work with selected param
3. After that everything is working

ps
diff
```javascript
git diff magento/module-catalog/view/adminhtml/templates/catalog/category/checkboxes/tree.phtml
diff --git a/magento/module-catalog/view/adminhtml/templates/catalog/category/checkboxes/tree.phtml b/magento/module-catalog/view/adminhtml/templates/catalog/category/checkboxes/tree.phtml
index b1f81976..af1b9eb1 100644
--- a/magento/module-catalog/view/adminhtml/templates/catalog/category/checkboxes/tree.phtml
+++ b/magento/module-catalog/view/adminhtml/templates/catalog/category/checkboxes/tree.phtml
@@ -73,7 +73,7 @@ Ext.extend(Ext.tree.TreePanel.Enhanced, Ext.tree.TreePanel, {
 jQuery(function()
 {
     var categoryLoader = new Ext.tree.TreeLoader({
-       dataUrl: '<?php /* @escapeNotVerified */ echo $block->getLoadTreeUrl() ?>'
+       dataUrl: '<?php /* @escapeNotVerified */ echo $block->getUrl('catalog/category_widget/categoriesJson') ?>'
     });
 
     categoryLoader.createNode = function(config) {
@@ -146,6 +146,7 @@ jQuery(function()
 
     categoryLoader.on("beforeload", function(treeLoader, node) {
         treeLoader.baseParams.id = node.attributes.id;
+        treeLoader.baseParams.selected = <?php /* @escapeNotVerified */ echo $block->getJsFormObject() ?>.updateElement.value;
     });
 
     categoryLoader.on("load", function(treeLoader, node, config) {
```

![screenshot from 2018-08-10 16-36-43](https://user-images.githubusercontent.com/813985/43960754-ae300480-9cbb-11e8-84d0-69f0677bf7ac.png)

**Comment by magento-engcom-team** [comment]
Hi @SKovbel. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me {$VERSION} instance` 

where `{$VERSION}` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (2.2-develop +). 
For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation.

@SKovbel do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by engcom-backlog-nickolas** [comment]
Hello @SKovbel, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by Bravarg** [comment]
Preparing PR

**Comment by magento-engcom-team** [comment]
Hi @SKovbel. Thank you for your report. 
The issue has been fixed in magento/magento2#18175 by @magently in 2.2-develop branch
Related commit(s):
 - [9f19592b3a1dc33a8a3a23c694880e5e6c5ca6a3](https://github.com/magento/magento2/commit/9f19592b3a1dc33a8a3a23c694880e5e6c5ca6a3)

The fix will be available with the upcoming 2.2.8 release.

**Comment by sidolov** [comment]
Hi @SKovbel. Thank you for your report. 
The issue has been fixed in magento/magento2#18500 by @gelanivishal in 2.3-develop branch
Related commit(s):
 - [cfc2339f563c68bc3593f058670e85a729bcf85c](https://github.com/magento/magento2/commit/cfc2339f563c68bc3593f058670e85a729bcf85c)

The fix will be available with the upcoming 2.3.1 release.
